samples/server/petstore/php-slim/lib/SlimRouter.php

```diff
@@ -586,6 +586,9 @@ public function __construct($container = [])
     {
         $this->slimApp = new App($container);
 
+        // middlewares requires Psr\Container\ContainerInterface instead of array
+        $container = $this->slimApp->getContainer();
+
```

Comment:
@ybelenko What about updating the method signature as well? e.g.
```
    public function __construct($appContainer = [])
    {
        $this->slimApp = new App($appContainer);
```

Reply:
@wing328 
Here is the thing, `new App(array $container = [])` constructor accepts array or `ContainerInterface`, but `App->getCongainer(): Psr\Container\ContainerInterface` method always returns `ContainerInnterface`. Slim Framework magic. :smile:

By the way, my biggest goal is to make specifically this class(`SlimRouter.php`) to be untoched by user. I think that file should be kind of library/internal, in perfect world/scenario user should not edit this file. When I will achieve this, comments and variable names will not be important anymore.

I try my best to store all user preferences and configs only within one file:
https://github.com/OpenAPITools/openapi-generator/blob/821fa40afc677523fdfb3781c3d48802f7236b56/samples/server/petstore/php-slim/index.php#L17-L28

Reply:
> Slim Framework magic

Understood. My suggestion is more on readability of the code. When I first read it, I didn't know why $container is reassigned. That's why I suggest naming the method argument differently to make sure aware that $container may not be the same as the $appContainer in the method argument.

Reply:
@wing328 
To make it more clear I would name it `$settings` or `$options` because input parameter technically it's not container yet. How is that sounds to you?

Reply:
Sounds good to me 👍